src/visualization/panel.cc

```diff
@@ -99,6 +100,19 @@ void VisualizationPanel::setRange(const size_t start, const size_t end) {
   sampler_->setRange(start, end);
 }
 
+bool VisualizationPanel::eventFilter(QObject *watched, QEvent *event) {
+  // filter out timer events for not visible visualisation so that we don't
+  // waste resources on rotating something that isn't visible.
+  if (!visible_ && event->type() == QEvent::Timer) {
```

Comment:
osobiscie to mi to nieprzeszkadza, ale kiedys na studiach mialem uciete punkty za konstruckje tego typu:
if (a) {return true;} else {return false;} zamiast return a;